atomicfu-transformer/src/main/kotlin/kotlinx/atomicfu/transformer/AtomicFUTransformer.kt

```diff
@@ -299,6 +308,66 @@ class AtomicFUTransformer(
         }
     }
 
+    // returns a type on which this is a potential accessor
+    private fun getPotentialAccessorType(access: Int, className: String, methodType: Type): Type? {
+        if (methodType.returnType !in AFU_TYPES) return null
+        return if (access and ACC_STATIC != 0) {
+            if (methodType.argumentTypes.size == 1 && methodType.argumentTypes[0].sort == OBJECT)
+                methodType.argumentTypes[0] else null
+        } else {
+            // if it not static, then it must be final
+            if (access and ACC_FINAL != 0 && methodType.argumentTypes.isEmpty())
+                Type.getObjectType(className) else null
+        }
+    }
+
+    private inner class ReferencesCollectorCV : CV(null) {
+        override fun visitMethod(access: Int, name: String, desc: String, signature: String?, exceptions: Array<out String>?): MethodVisitor? {
+            return ReferencesCollectorMV(className, access, name, desc, signature, exceptions)
+        }
+    }
+
+    private inner class ReferencesCollectorMV(
+        private val className: String,
+        access: Int, name: String, desc: String, signature: String?, exceptions: Array<out String>?
+    ) : MethodNode(ASM5, access, name, desc, signature, exceptions) {
+
+        override fun visitEnd() {
+            // skip accessor method - already registered in the previous phase
+            val methodType = Type.getMethodType(desc)
+            getPotentialAccessorType(access, className, methodType)?.let {
+                return
+            }
+            // not accessor
+            var i = instructions.first
+            while (i != null) {
+                i = getPotentialExternalAccessorInvokes(i)
+            }
+        }
+
+        private fun getPotentialExternalAccessorInvokes(i: AbstractInsnNode): AbstractInsnNode? {
+            if (i is MethodInsnNode) {
+                val methodId = MethodId(i.owner, i.name, i.desc, i.opcode)
+                val getter = methodId.name
+                // compare owner packages
+                if (methodId.owner.substringBeforeLast('/') != className.substringBeforeLast('/')) {
+                    val fieldName = toFieldName(getter).decapitalize()
```

Comment:
`toFieldName` is fragile (depends on compiler method mangling that may change) and is not needed at all. You have top-level map `accessors` that was build on the 1st phase. A method invocation is an accessor if and only if it is was stored in `accessors`.